src/stk/itk_bridge/itk_io.cpp

```diff
@@ -0,0 +1,295 @@
+#include "itk_io.h"
+#include "itk_convert.h"
+
+#include <itkImageFileReader.h>
+#include <itkImageFileWriter.h>
+#include <itkImageIOBase.h>
+
+
+/*!
+ * \brief Read a specific image type with ITK.
+ */
+template<class ImageType>
+stk::Volume read(const itk::ImageIOBase::Pointer image_io)
+{
+    auto reader = itk::ImageFileReader<ImageType>::New();
+    reader->SetImageIO(image_io);
+    reader->SetFileName(image_io->GetFileName());
```

Comment:
It seems it is possible to pass an existing ImageIO to the reader, but the file name must be passed too. I am not sure how much this will help to prevent redundant file operations, depends pretty much on the implementation of the single file readers, which in turn depend on the backing format libraries.